backend/utills/http.utills.ts

```diff
@@ -0,0 +1,34 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { allPublishedProfilesPayload } from '../profile/profile.service'
+
+export async function allRequestController(
+  req: NextApiRequest,
+  callbackFn: () => Promise<void>,
+) {
+  switch (req.method) {
```

Comment:
**issue:** Problem with this function is that you have the same handler to the different methods, and you won't be able to run this for handling multiply http methods. You should fix it similar like example below:

```js
export async function allRequestController(
  req: NextApiRequest,
  res: NextApiResponse,
  httpHandlers: {
    POST: (req: NextApiRequest, res: NextApiResponse) => Promise<void>
    GET: (req: NextApiRequest, res: NextApiResponse) => Promise<void>
  }
) {
  switch (req.method) {
    case 'GET':
      return httpHandlers.GET(req, res)
    case 'POST':
      return httpHandlers.POST(req, res)
    default:
      res.status(405).end(`Method ${req.method} Not Allowed`)
  }
}
```